### ISSUE TITLE
setAppleSearchAdsDebugMode not working

### ISSUE THREAD

**Issue by mhemmings** [BranchMetrics/ios-branch-deep-linking#682]
We have implemented `setAppleSearchAdsDebugMode` exactly as mentioned in the docs and running on a clean simulator (10.3.1). The same behaviour also happens on a device.

When trying to log the dummy debug data, nothing is getting returned other than the default:
```
{
    "+clicked_branch_link" = 0;
    "+is_first_session" = 0;
}
```

Our logging code:
```
[branch initSessionWithLaunchOptions:launchOptions isReferrable:YES andRegisterDeepLinkHandler:^(NSDictionary *params, NSError *error) {
        NSLog(@"data: %@", [params description]);
}];
```

In addition, no attribution appears in the Branch dashboard

**Comment by E-B-Smith** [comment]
Testing Apple Search Ads can be tricky. Make sure that:

1. You've included the iAd and AdSupport frameworks in your project.

2. We only check for Apple Search ad attribution the first 30 days from the day the app was installed.  You can simulate this by removing the app from your phone or simulator, then re-running the app.  Branch will delay the init and check for Apple Search ad attribution.

Let me know if this works for you!

**Comment by mhemmings** [comment]
Haven't had chance to check this as our iOS lead is away. I'll reopen if we carry on seeing the same thing 😃

**Comment by E-B-Smith** [comment]
Okay, please do. 

I'm able to get the test Search Ads results if I follow the steps above.

**Comment by mhemmings** [comment]
So we've checked the project and can confirm the frameworks are added, yet still nothing appears in the innit session data

**Comment by mhemmings** [comment]
@E-B-Smith Any ideas on why this isn't working?

**Comment by E-B-Smith** [comment]
Did you check this too?

> 2. We only check for Apple Search ad attribution the first 30 days from the day the app was installed. You can simulate this by removing the app from your phone or simulator, then re-running the app. Branch will delay the init and check for Apple Search ad attribution.

**Comment by mhemmings** [comment]
Yep we did.
Though this shouldn't effect it as stated in the docs:
>In this mode fake campaign params are returned 100% of the time

**Comment by E-B-Smith** [comment]
Does it work for you when you build and run in the Branch-TestBed in Simulator?

Here's how I initialize Branch in the TestBed to get the Apple Search Ads link:

<img width="771" alt="screen shot 2017-08-25 at 9 11 39 am" src="https://user-images.githubusercontent.com/1229047/29722447-ae809bde-8975-11e7-84bf-00d15347a73d.png">

**Comment by mhemmings** [comment]
Interestingly, the testbed app does not work either, with the same config as you have above. The params logged are the same:
```
{
    "+clicked_branch_link" = 0;
    "+is_first_session" = 1;
}
```
and nothing more

**Comment by E-B-Smith** [comment]
What country / region are you in?

This is what I get:
```
{
    "$3p" = "a_apple";
    "+click_timestamp" = 1503677443;
    "+clicked_branch_link" = 1;
    "+is_first_session" = 1;
    "+match_guaranteed" = 1;
    "~ad_set_id" = 1234567890;
    "~ad_set_name" = AdGroupName;
    "~branch_ad_format" = "App Only";
    "~campaign" = CampaignName;
    "~campaign_id" = 1234567890;
    "~channel" = "Apple App Store";
    "~creative_name" = CreativeName;
    "~feature" = "Search Ads";
    "~referring_link" = "https://bnctestbed.app.link?%243p=a_apple&%7Esecondary_ad_format=Apple+Search+Ads&%7Etags=%5BKeyword%2C+LineName%5D&%7Ebranch_ad_format=App+Only&%7Ecampaign=CampaignName&%7Ead_set_id=1234567890&%7Echannel=Apple+App+Store&%7Efeature=Search+Ads&%7Ecreative_name=CreativeName&%7Ead_set_name=AdGroupName&%7Ecampaign_id=1234567890";
    "~secondary_ad_format" = "Apple Search Ads";
    "~tags" =     (
        Keyword,
        LineName
    );
}
```

These are the libraries I link with:

<img width="878" alt="screen shot 2017-08-25 at 9 36 40 am" src="https://user-images.githubusercontent.com/1229047/29723309-f366bc1c-8978-11e7-8bd2-ec354295350b.png">

**Comment by mhemmings** [comment]
We're in the UK.
Test bed app has the exact same linked libraries as above.

**Comment by E-B-Smith** [comment]
I'm out of ideas at the moment. If you zip up and email me you project I'm happy to look at it.

**Comment by E-B-Smith** [comment]
Did you make any progress on this?

**Comment by mhemmings** [comment]
@E-B-Smith Nope, still got an empty response :(

**Comment by mhemmings** [comment]
We can't send our project unfortunately, but if the testbed app pulled down from here isn't working for us then that doesn't really matter. FWIW, our lead iOS dev sees this same behaviour as I do too (on our project and the testbed app)

I've put breakpoints in the `checkAppleSearchAdsAttribution` method and can confirm that the `testInfo` object is set up as expected (https://github.com/BranchMetrics/ios-branch-deep-linking/blob/03ff243248299aafb98e48f075181a5d64efe651/Branch-SDK/Branch-SDK/Branch.m#L793)
but it just doesn't get returned to the listener.

**Comment by E-B-Smith** [comment]
Set a break point on line 789 of the same file.  If Apple *does* return Apple Search Ad info then we use that data.  Otherwise we use the test data we dummy up in the `testInfo` dictionary.

When Apple Search Ads first started, Apple returned Search Ad data erratically, so I added the `testInfo` dictionary to dummy up the data so the rest of the system could be tested.

<img width="1016" alt="screen shot 2017-08-29 at 11 20 10 am" src="https://user-images.githubusercontent.com/1229047/29837222-6f51403c-8cac-11e7-81cd-eaae1d6a644a.png">

<img width="441" alt="screen shot 2017-08-29 at 11 16 14 am" src="https://user-images.githubusercontent.com/1229047/29837244-78f96a38-8cac-11e7-96ef-2d94f1de9496.png">

**Comment by mhemmings** [comment]
Nothing in `attrDetails`, and initSession still doesn't get the dummy data

![screen shot 2017-08-30 at 09 08 02](https://user-images.githubusercontent.com/5692567/29861935-e8c060e0-8d62-11e7-9231-d6a758f9bf61.png)

**Comment by E-B-Smith** [comment]
I am digging in to this. Obviously something is weird.

**Comment by mhemmings** [comment]
OK, thanks for your help!! 👍

**Comment by mhemmings** [comment]
Hey @E-B-Smith, any luck on this? See you've been trying some things, hope it's not been too painful!

**Comment by E-B-Smith** [comment]
I found a problem with the back end.  That code is in production now. I am going to do some final testing today and you should be good to go.

**Comment by E-B-Smith** [comment]
I fixed this in version 0.18.8.  

https://github.com/BranchMetrics/ios-branch-deep-linking/releases/tag/0.18.8
